app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,55 @@
+class UsersController < ApplicationController
+
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect "/tweets/#{User.find_by(username: session[:username]).slug}}"
+    else
+      erb :"/sessions/signup"
+    end
+  end
+
+    get '/login' do
+      if logged_in?
+        redirect "/tweets"
+      else
+        erb :"/sessions/login"
+      end
+    end
+
+    get '/logout' do
+      session.clear
+      redirect '/login'
+    end
+
+    get '/users/:username' do
+      @user = User.find_by(params[:user])
+      erb :"/tweets/show"
+    end
+
+  post '/signup' do
+    if params[:username]=="" || params[:email]=="" || params[:password]==""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.